src/briefcase/commands/base.py

```diff
@@ -324,104 +337,102 @@ def binary_path(self, app):
 
         :param app: The app config
         """
-        ...
 
-    def _load_path_index(self, app: BaseConfig):
-        """Load the path index from the index file provided by the app template.
+    def briefcase_toml(self, app: AppConfig) -> dict[str, ...]:
+        """Load the ``briefcase.toml`` file provided by the app template.
 
         :param app: The config object for the app
-        :return: The contents of the application path index.
+        :return: The contents of ``briefcase.toml``
         """
         try:
-            with (self.bundle_path(app) / "briefcase.toml").open("rb") as f:
-                self._path_index[app] = tomllib.load(f)["paths"]
-        except OSError as e:
-            raise BriefcaseCommandError(
-                f"Unable to find '{self.bundle_path(app) / 'briefcase.toml'}'"
-            ) from e
-        return self._path_index[app]
+            return self._briefcase_toml[app]
+        except KeyError:
+            try:
+                with (self.bundle_path(app) / "briefcase.toml").open("rb") as f:
+                    self._briefcase_toml[app] = tomllib.load(f)
+            except OSError as e:
+                raise MissingAppMetadata(self.bundle_path(app)) from e
+            else:
+                return self._briefcase_toml[app]
```

Comment:
FYI...I realized something about this coding style while testing this.

When Rich prints the `MissingAppMetadata` exception to the log file, it doesn't actually roll back up stack of exceptions in to the true source of the exception (that is, whatever code is trying to read `briefcase.toml`). If, however, you raise the `MissingAppMetadata` from the `KeyError` exception, then Rich includes the full stack in the exception print out.

I'm not sure if I think that's a good way to mitigate this problem....but it's something to thing about for this coding style.